pkg/kusionctl/cmd/apply/options.go

```diff
@@ -150,6 +151,17 @@ func (o *ApplyOptions) Run() error {
 		}
 	}
 
+	// Replace ref secrets if needed
+	secure, ok := stack.Labels[projectstack.LabelSecure].(bool)
+	if ok && secure {
+		if !project.SecretStores.IsValid() {
+			return fmt.Errorf("no secret store is provided")
+		}
+		if err = vals.ReplaceRefs(planResources, project.SecretStores); err != nil {
```

Comment:
I try to merge this logic into node execute(), it looks like not quite suitable, since secret ref need a secured label marking there is one or many secret refs need to be restore. 
And in the restoring logic, need secretStore definition, which is also required, so the func definition is changed, and a lot variables need to passed.